docs/src/main/sphinx/connector/delta-lake.md

```diff
@@ -716,6 +716,46 @@ The output of the query has the following history columns:
   - Whether or not the operation appended data
 :::
 
+##### `$partitions` table
+
+The `$partitions` table provides a detailed overview of the partitions of the
+Delta Lake table.
+
+You can retrieve the information about the partitions of the Delta Lake table
+`test_table` by using the following query:
+
+```
+SELECT * FROM "test_table$partitions"
+```
+
```

Comment:
I was looking over what Iceberg is doing to expose `record_count` and `data` columns 
https://trino.io/docs/current/connector/iceberg.html#partitions-table

In case that the `add.stats` would be missing in `add` file from a partition, the `record_count`  column or the `data`  fields would get invalidated (set to `NULL`).
relevant Iceberg logic to invalidate a field in case of not having stats present:
https://github.com/trinodb/trino/blob/9606268002e1526e8b61b2231a85ff604edee7c7/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java#L314-L333

The outcome of having this extra information is that it would give to the user a good overview on what is actually contained in the partition besides rather generic content like `file_count` or the `total_size` of the files.

Reply:
Does it make sense to address this in a separate issue/PR? I could see this feature potentially requiring a significant amount of code/tests.